File: mcl-app/src/main/java/com/mcl/mancala/beans/PlayerMove.java
package com.mcl.mancala.beans;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


/*
 * Bean used for handling the game movements that are done from front end side
 *
 * @author Maxim N
 * */

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class PlayerMove {
    private Mancala mancala;
    private int playerToMove;
    private int smallPitToMoveFrom;
}

File: mcl-ui/angular-mancala/src/app/models/PlayerMove.ts
import { Mancala } from "./Mancala";

export class PlayerMove {
  constructor(
    id: number,
    playerToMove: number,
    smallPitToMoveFrom: number,
    mancala?: Mancala
  ) {
    this.id = id;
    this.playerToMove = playerToMove;
    this.smallPitToMoveFrom = smallPitToMoveFrom;
    this.mancala = mancala;
  }
  id: number;
  playerToMove: number;
  smallPitToMoveFrom: number;
  mancala: Mancala;
}
